ezGroceriesShoppingList/src/main/java/com/ezgroceries/shoppinglist/web/WebController.java

```diff
@@ -0,0 +1,74 @@
+package com.ezgroceries.shoppinglist.web;
+
+import com.ezgroceries.shoppinglist.entity.Cocktail;
+import com.ezgroceries.shoppinglist.entity.ShoppingList;
+import com.ezgroceries.shoppinglist.service.CocktailService;
+import com.ezgroceries.shoppinglist.service.ShoppingListService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+
+public class WebController {
+
+    private static final Logger log = LoggerFactory.getLogger(WebController.class);
+
+
+    @Autowired
+    private CocktailService cocktailService;
+    @Autowired
+    private ShoppingListService shoppingListService;
+
+    @GetMapping("/cocktails")
+    @ResponseBody
+    public List<Cocktail> getCocktails(@RequestParam String search) {
+        List<Cocktail> cocktails = cocktailService.listCocktails(search);
+        log.info("List of cocktails compiled");
+        return cocktails;
+
+    }
+
+    @PostMapping("/shopping-list")
+    public ResponseEntity<ShoppingList> createShoppingList(@RequestBody ShoppingList shoppingList) {
+        log.info("Successfully created a shopping list");
+        return buildResponseEntity(shoppingList);
+    }
+
+    @GetMapping("/shopping-lists/{shoppingListId}")
+    @ResponseBody
+    public ShoppingList findShoppingListById(@PathVariable("shoppingListId") String shoppingListId) {
+        ShoppingList shoppingList = shoppingListService.findShoppingListById(shoppingListId);
+        return shoppingList;
+    }
+
+    @PostMapping("/shopping-lists/{shoppingListId}/cocktails")
+    public ResponseEntity<ShoppingList> addCocktailToShoppingList(@RequestBody Cocktail cocktail, @PathVariable("shoppingListId") String shoppingListId) {
+        String cocktailId = cocktail.getCocktailId();
```

Comment:
There is no need to extract this coctailId in another variable, remove this line please